source/common/http/header_map_impl.h

```diff
@@ -18,46 +18,36 @@ namespace Http {
+    appendInline(HeaderHandles::get().name, data, delimiter);                                      \
   }                                                                                                \
   void setReference##name(absl::string_view value) override {                                      \
-    HeaderEntry& entry = maybeCreateInline(&inline_headers_.name##_, Headers::get().name);         \
-    updateSize(entry.value().size(), value.size());                                                \
-    entry.value().setReference(value);                                                             \
+    setReferenceInline(HeaderHandles::get().name, value);                                          \
   }                                                                                                \
   void set##name(absl::string_view value) override {                                               \
-    HeaderEntry& entry = maybeCreateInline(&inline_headers_.name##_, Headers::get().name);         \
-    updateSize(entry.value().size(), value.size());                                                \
-    entry.value().setCopy(value);                                                                  \
+    setInline(HeaderHandles::get().name, value);                                                   \
   }                                                                                                \
-  void set##name(uint64_t value) override {                                                        \
-    HeaderEntry& entry = maybeCreateInline(&inline_headers_.name##_, Headers::get().name);         \
-    subtractSize(inline_headers_.name##_->value().size());                                         \
-    entry.value().setInteger(value);                                                               \
-    addSize(inline_headers_.name##_->value().size());                                              \
-  }                                                                                                \
-  size_t remove##name() override { return removeInline(&inline_headers_.name##_); }
-
-#define DEFINE_INLINE_HEADER_STRUCT(name) HeaderEntryImpl* name##_;
+  void set##name(uint64_t value) override { setInline(HeaderHandles::get().name, value); }         \
+  size_t remove##name() override { return removeInline(HeaderHandles::get().name); }               \
+  absl::string_view get##name##Value() const override {                                            \
+    return getInlineValue(HeaderHandles::get().name);                                              \
+  }
 
 /**
  * Implementation of Http::HeaderMap. This is heavily optimized for performance. Roughly, when
- * headers are added to the map, we do a hash lookup to see if it's one of the O(1) headers.
+ * headers are added to the map, we do a trie lookup to see if it's one of the O(1) headers.
```

Comment:
suggest: s/added to the map/added to the map by string/ and clarify that when added to the map via direct generated methods that no such trie lookup is needed?